### PR TITLE
Find service area interactive

### PR DESCRIPTION
**Pull request by singhgag**

### CODE REVIEW

File: arcgis-runtime-samples-macos/Route & Directions/Find service area interactive/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+## How to use the sample
+
+Use the segmented control in the toolbar to switch between `Facilities` and `Barriers`. To add a facility click on a location on map and a marker will be added. For barrier, click on the map to draw buffered polygon. You can use the sliders to change the time breaks. Then click the `Service Area` button to get the service area for added facilities. Click on the clear button (trash) to start over.
+
+![](image1.png)
+
+## How it works
+
+The sample uses the `defaultRouteParameters(completion:)` method on `AGSServiceAreaTask` to get the default parameters from the service. Barriers are created using the initializer `init(polygon:)` on `AGSPolygonBarrier`. Sets the facilities and barriers in the parameters. Then uses the `solveServiceArea(with:completion:)` method to solve for the route. Once the result is in, the sample displays the service areas for individual facilities using `resultPolygons(atFacilityIndex:)` method on `AGSServiceAreaResult`.
```
Comment (dg0yal) [inline arcgis-runtime-samples-macos/Route & Directions/Find service area interactive/README.md]: defaultRouteParameters() needs to be changed for service area

Comment (dg0yal) [inline arcgis-runtime-samples-macos/Route & Directions/Find service area interactive/README.md]: Just noticed same problem exists for iOS. You might want to make the change at both places

File: arcgis-runtime-samples-macos/Route & Directions/Find service area interactive/FindServiceAreaInteractiveVC.swift [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        
+        //solve for service area
+        self.serviceAreaTask.solveServiceArea(with: self.serviceAreaParameters) { [weak self] (result: AGSServiceAreaResult?, error: Error?) in
+            
+            guard let weakSelf = self else {
+                return
+            }
+            
+            guard error == nil else {
+                print("Error solving service area:: \(error!.localizedDescription)")
```
Comment (dg0yal) [inline arcgis-runtime-samples-macos/Route & Directions/Find service area interactive/FindServiceAreaInteractiveVC.swift]: Can you display this to the user instead? I ran into a case where it failed when i added a facility outside of the network area. But it never worked and I didn't know the op failed

File: arcgis-runtime-samples-macos/Route & Directions/Find service area interactive/README.md (hunk 1 above)
Comment (singhgag) [inline arcgis-runtime-samples-macos/Route & Directions/Find service area interactive/README.md]: @dg0yal the method is asynchronous thats why the completion parameter.

**Comment by singhgag** [comment]
@dg0yal I have made the changes. Please take another look.

File: arcgis-runtime-samples-macos/Route & Directions/Find service area interactive/README.md (hunk 1 above)
Comment (dg0yal) [inline arcgis-runtime-samples-macos/Route & Directions/Find service area interactive/README.md]: i guess i wasn't clear.
It currently says "Route" parameters. Looks like a copy paste error. Should be "Service Area" parameters

**Comment by singhgag** [comment]
Done.

**Pull request merged**